CodeEditModules/Modules/Git/src/Client/Models/ChangedFiles.swift

```diff
@@ -0,0 +1,138 @@
+//
+//  ChangedFiles.swift
+//  
+//
+//  Created by Nanashi Li on 2022/05/05.
+//
+
+import Foundation
+import SwiftUI
+
+public struct ChangedFiles: Codable, Hashable, Identifiable {
```

Comment:
Try to merge this struct with the existing `FileItem`. The thing is that the two structs share a lot of code. For example if we decide to change the color of one file extension we need to change it in both files. Try to implement your needed properties in `FileItem` directly to minimise code duplication.

Reply:
Why not just use extension FileItem

Reply:
What do you mean?

Reply:
I think we can extract their shared portions like `fileIcon` and `iconColor` to be static functions for reusability.

Reply:
> I think we can extract their shared portions like `fileIcon` and `iconColor` to be static functions for reusability.

Tried that in a way but couldn't access it in the workspace client... Any ideas?